src/opnsense/mvc/app/views/OPNsense/CheckIP/index.volt

```diff
@@ -0,0 +1,209 @@
+
+  $("#grid-services").UIBootgrid({
+    'search':'/api/checkip/settings/search/service/',
+    'get':'/api/checkip/settings/get/service/',
+    'set':'/api/checkip/settings/set/service/',
+    'add':'/api/checkip/settings/set/service/',
+    'del':'/api/checkip/settings/del/service/',
+    'toggle':'/api/checkip/settings/toggle/service/'
+  });
+
+  // Determine if at least one service is enabled as the default.  Warn if not.
+  function is_service_default() {
+    ajaxCall(url="/api/checkip/settings/is_default/service/", sendData={}, callback=function(data,status) {
+      if (status == 'success' && data['result']) {
+         $("#responseMsg").css({'background-color': '', 'color': ''});
+         $("#responseMsg").addClass('hidden');
+         $("#responseMsg").html('');
+      } else {
+         $("#responseMsg").html('<b>WARNING: </b>There is no default check IP service enabled!');
+         $("#responseMsg").css({'background-color': 'yellow', 'color': 'red'});
+         $("#responseMsg").removeClass('hidden');
+      }
+    });
+  }
+
+  // Apply changes to the DOM
+  function dom_update() {
+    // suppress FDS selection checkbox and command buttons, and keep unselected
+    $('table#grid-services tbody tr[data-row-id="FDS"]').each(function() {
+      $(this).find('td.select-cell input.select-box').addClass('hidden');
+      $(this).find('td button').filter('.command-edit,.command-copy,.command-delete').addClass('hidden');
+/**/
+      $(this).find('td.select-cell input.select-box').prop('checked', false);
+      $(this).attr('aria-selected', false);
+      $(this).removeClass('active');
+/**/
+    });
+
+    // suppress fa-times icon for verify ssl peer disabled
+    $('table#grid-services tbody tr td:nth-child(5) span').removeClass('fa-times');
+
+    // hyper-link each service URL
+    $('table#grid-services tbody tr td:nth-child(4)').each(function() {
+      $(this).html('<a target="Check_IP" rel="noopener noreferrer" href="'+$(this).text()+'">'+$(this).text()+'</a>');
+    });
+  }
+
+  // DOM observer
+  var targetNode = document.querySelector('table#grid-services tbody');
+  var config = { childList: true };
+  var callback = function() {
+    dom_update();
+    is_service_default();
+  };
+  var observer = new MutationObserver(callback);
+  observer.observe(targetNode, config);
+
+  $('div#services').removeClass('hidden');
+
+
+  /**
+   * service test
+   */
+
+  $('[data-toggle][href=#test]').click(function(){
+    $('#full_help_button').addClass('hidden');
+    var data_get_map = {'frm_TestSettings':"/api/checkip/test/get/service/"};
+    mapDataToFormUI(data_get_map).done(function(data){
+      // place actions to run after load, for example update form styles.
+        formatTokenizersUI();
+        $('.selectpicker').selectpicker('refresh');
+        // request service status on load and update status box
+        updateServiceControlUI('checkip');
+    });
+  });
+
+  $('#btn_test').unbind('click').click(function(){
```

Comment:
this function is deprecated (v3.0), use off instead: https://api.jquery.com/unbind/